planner/core/expression_rewriter.go

```diff
@@ -1631,35 +1631,53 @@ func (er *expressionRewriter) patternLikeToExpression(v *ast.PatternLikeExpr) {
 	var function expression.Expression
 	fieldType := &types.FieldType{}
 	isPatternExactMatch := false
+	escape := "\\"
+	if escapeExpression, ok := er.ctxStack[l-1].(*expression.Constant); ok {
+		str, isNull, err := escapeExpression.EvalString(nil, chunk.Row{})
+		if err != nil {
+			er.err = err
+			return
+		}
+		if !isNull {
+			escape = str
+		}
+	}
 	// Treat predicate 'like' the same way as predicate '=' when it is an exact match and new collation is not enabled.
-	if patExpression, ok := er.ctxStack[l-1].(*expression.Constant); ok && !collate.NewCollationEnabled() {
+	isPreparedParamMarker := false
+	switch n := v.Escape.(type) {
+	case *driver.ParamMarkerExpr:
+		if !n.InExecute {
```

Comment:
this file is all about transforming `AST` to `expression`, which will only happen in the build plan step, and `InExecute` is used in runtime.

Here, we should only consider the `non-paramaker` scenario, so we will try to rewrite `like` to `eq` if the escape type is `expression.constant`.

Reply:
Understood! I changed it as should already